test_suite/tests/derive.rs

```diff
@@ -140,14 +154,34 @@ fn recursive_type_derive() {
 
 	let ty = Type::builder().path(Path::new("Tree", "derive")).variant(
 		Variants::with_fields()
-			.variant("Leaf", Fields::named().field_of::<i32>("value"))
+			.variant(
+				"Leaf",
+				Fields::named().field(Field::named_of::<i32>("value").with_type_display_name(vec!["i32"])),
+			)
 			.variant(
 				"Node",
 				Fields::named()
-					.field_of::<Box<Tree>>("right")
-					.field_of::<Box<Tree>>("left"),
+					.field(Field::named_of::<Box<Tree>>("right").with_type_display_name(vec!["Box"]))
+					.field(Field::named_of::<Box<Tree>>("left").with_type_display_name(vec!["Box"])),
 			),
 	);
 
 	assert_type!(Tree, ty);
 }
+
+#[test]
+fn fields_with_type_alias() {
+	type BoolAlias = bool;
```

Comment:
Unrelated to this PR, but do you think we could have this work at some point?

```
#[derive(TypeInfo)]
type BoolAlias = bool;
```

Reply:
It's a good idea, it may well be worth experimenting with. This would provide extra information about what aliases are available and may increase the accuracy of the heuristic identifying alias usage. 

Ultimately it would also have the same limitation in that there is still no guarantee to identify when a given alias is being used, because the registry is using the underlying type id i.e. `any::TypeId::of<BoolAlias>() == any::TypeId::of<bool>()`.

Reply:
Never thought about this use case. I am not sure it goes too far but making it optional would probably be okay. The question is how to integrate it properly into the current scheme if we actually want ot.